jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannelOverHttp.java

```diff
@@ -61,6 +62,15 @@
     private boolean _expect102Processing = false;
     private List<String> _complianceViolations;
     private HttpFields.Mutable _trailers;
+    // Both _content and _specialContent don't need to be volatile nor protected by a lock
+    // as they're always accessed by the same thread, i.e.: we get notified by onFillable
+    // that the socket contains new bytes and either schedule an onDataAvailable
+    // call that is going to read the socket or release the blocking semaphore to wake up
+    // the blocked reader and make it read the socket. The same logic is true for async
+    // events like timeout: we get notified and either schedule onError or release the
+    // blocking semaphore.
+    private HttpInput.Content _content;
+    private HttpInput.Content _specialContent;
```

Comment:
My original analysis was that `_specialContent` is only ever written by the scheduler thread, then either notifies the blocked thread with the semaphore (which gives us a _happens-before_) or reschedules a thread that reads it. A job submission to a standard `ExecutorService` creates a _happens-before_ relationship that would make this safe but I may need to check if the form of scheduling we use in EWYK does too. I suspect it does, but I'll review this in more details.

Reply:
Looks safe to me, but I could mark `_specialContent` as volatile nevertheless. This would make the code a bit less brittle in wrt potential future changes.

Reply:
why do we need both `_content` and `_specialContent`.  We only have a single content in H2 and I think similar logic could be applied here.

Reply:
I got rid of _specialContent.